micro-rdk/src/esp32/conn/network.rs

```diff
@@ -202,25 +202,30 @@ impl Esp32WifiNetwork {
 
         let sl_stack = esp32_get_system_event_loop()?;
 
-        let subscription = sl_stack.subscribe::<WifiEvent, _>(move |event: WifiEvent| {
-            if matches!(event, WifiEvent::StaDisconnected(_)) {
-                if let Ok(wifi) = esp32_get_wifi() {
-                    if let Some(mut wifi_guard) = wifi.try_lock() {
-                        let wifi_mut = wifi_guard.wifi_mut();
-                        if let Err(err) = wifi_mut.connect() {
-                            let ssid = wifi_mut
-                                .get_configuration()
-                                .map_or("<no_ssid>".to_owned(), |c| {
-                                    c.as_client_conf_ref().unwrap().ssid.to_string()
-                                });
-                            log::error!("could not connect to WiFi {} cause : {:?}", ssid, err);
+        let subscription =
+            sl_stack.subscribe::<WifiEvent, _>(move |event: WifiEvent| match event {
+                WifiEvent::StaDisconnected(disconnected) => {
+                    let reason: super::wifi_error::WifiErrReason = disconnected.reason().into();
+                    log::info!("StaDisconnected event received: {:?}", reason);
```

Comment:
From your slack message, I can see that logging the reason with `{:?}` is going to look like `GroupKeyUpdateTimeout(16)`. However, the 16 isn't really helpful there, maybe. It makes it look like the 16 is somehow parameterizing the `GroupKeyUpdateTimeout`, but it is really just the underlying code. Maybe `Debug` on `WifiErrReason` isnt' the right way to log it.